source/server/hot_restarting_child.cc

```diff
@@ -154,11 +155,33 @@ void HotRestartingChild::registerUdpForwardingListener(
   udp_forwarding_context_.registerListener(address, listener_config);
 }
 
+void HotRestartingChild::registerParentDrainedCallback(
+    const Network::Address::InstanceConstSharedPtr& address, absl::AnyInvocable<void()> callback) {
+  if (parent_drained_) {
+    callback();
+  } else {
+    auto result = on_drained_actions_.try_emplace(address->asString(), std::move(callback));
+    RELEASE_ASSERT(result.second,
+                   fmt::format("two udp listeners with the same address ({}) is unexpected",
+                               address->asString()));
+  }
+}
+
+void HotRestartingChild::allDrainsImplicitlyComplete() {
+  for (auto& drain_action : on_drained_actions_) {
+    // call the callback.
+    std::move(drain_action.second)();
```

Comment:
What does the `std::move` do here? Is it needed?

Reply:
Yeah, this is required for calling an `AnyInvocable` to do with it being a "call once" type.